subprojects/model-core/src/main/java/org/gradle/internal/instantiation/AbstractClassGenerator.java

```diff
@@ -857,19 +859,21 @@ boolean claimPropertyImplementation(PropertyMetadata property) {
             }
 
             // Property is readable and all getters and setters are abstract
-
             if (property.setters.isEmpty()) {
                 if (property.getType().equals(ConfigurableFileCollection.class)
                     || property.getType().equals(ListProperty.class)
                     || property.getType().equals(SetProperty.class)
                     || property.getType().equals(MapProperty.class)
                     || property.getType().equals(RegularFileProperty.class)
                     || property.getType().equals(DirectoryProperty.class)
-                    || property.getType().equals(Property.class)) {
+                    || property.getType().equals(Property.class)
+                    || property.getMainGetter().method.getAnnotation(Nested.class) != null
```

Comment:
do you think it would make sense to use a `Set<Class<?>>#contains` here instead of serial equality check?

Reply:
That may make it easier to read. I think it is good enough to read as it is currently, and I want to touch as little of the class generation code as possible, so I would leave it as is.

Reply:
I ended up using a `Set`.